sklearn/preprocessing/_encoders.py

```diff
@@ -0,0 +1,845 @@
+# Authors: Andreas Mueller <amueller@ais.uni-bonn.de>
+#          Joris Van den Bossche <jorisvandenbossche@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import division
+
+import numbers
+import warnings
+
+import numpy as np
+from scipy import sparse
+
+from ..base import BaseEstimator, TransformerMixin
+from ..externals import six
+from ..utils import check_array
+from ..utils import deprecated
+from ..utils.fixes import _argmax
+from ..utils.validation import check_is_fitted, FLOAT_DTYPES
+from .label import LabelEncoder
+
+
+range = six.moves.range
+
+
+__all__ = [
+    'OneHotEncoder',
+    'OrdinalEncoder'
+]
+
+
+def _transform_selected(X, transform, selected="all", copy=True):
```

Comment:
Shouldn't this be shared with other preprocessing?

Reply:
No, OneHotEncoder is the only transformer that uses this (that has the ability to only transform a subset of the columns)